src/gui/tray/UserLine.qml

```diff
@@ -36,10 +36,11 @@ MenuItem {
             Accessible.role: Accessible.Button
             Accessible.name: qsTr("Switch to account") + " " + name
 
-            onClicked: if (!isCurrentUser) {
-                UserModel.switchCurrentUser(id)
-            } else {
-                accountMenu.close()
+            onClicked: {
+                if (!model.isCurrentUser) {

```

Comment:
should we not just do `UserModel.currentUserId != id` or even not check if it is current given the setter should be noop in case we call it with its current value